File: src/main/java/ru/stnk/RestTestAPI/repository/RolesRepository.java
package ru.stnk.RestTestAPI.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.stnk.RestTestAPI.entity.Roles;

@Repository
public interface RolesRepository extends JpaRepository<Roles, Long> {

    Roles findByName(String name);

}

File: src/main/kotlin/ru/stnk/resttestapi/repository/RolesRepository.kt
package ru.stnk.resttestapi.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.stnk.resttestapi.entity.Role
import ru.stnk.resttestapi.entity.RoleName

@Repository
interface RolesRepository : JpaRepository<Role, Long> {

    fun findByName(roleName: RoleName): Role

}
